packages/editor/src/components/list-view-sidebar/index.js

```diff
@@ -22,29 +23,63 @@ import { store as editorStore } from '../../store';
 
 const { TabbedSidebar } = unlock( blockEditorPrivateApis );
 
+// Used to count how many times the component renders and determine the initial focus logic.
+let renderCounter = 0;
+
 export default function ListViewSidebar() {
 	const { setIsListViewOpened } = useDispatch( editorStore );
-	const { getListViewToggleRef } = unlock( useSelect( editorStore ) );
+
+	const { listViewToggleRef, showListViewByDefault } = useSelect(
+		( select ) => {
+			const { getListViewToggleRef } = unlock( select( editorStore ) );
+			const _showListViewByDefault = select( preferencesStore ).get(
+				'core',
+				'showListViewByDefault'
+			);
+
+			return {
+				listViewToggleRef: getListViewToggleRef(),
+				showListViewByDefault: _showListViewByDefault,
+			};
+		},
+		[]
+	);
 
 	// This hook handles focus when the sidebar first renders.
 	const focusOnMountRef = useFocusOnMount( 'firstElement' );
 
 	// When closing the list view, focus should return to the toggle button.
 	const closeListView = useCallback( () => {
 		setIsListViewOpened( false );
-		getListViewToggleRef().current?.focus();
-	}, [ getListViewToggleRef, setIsListViewOpened ] );
+		listViewToggleRef.current?.focus();
+	}, [ listViewToggleRef, setIsListViewOpened ] );
 
 	const closeOnEscape = useCallback(
 		( event ) => {
 			if ( event.keyCode === ESCAPE && ! event.defaultPrevented ) {
+				// Always use `event.preventDefault` before calling `closeListView`.
+				// This is important to prevent the `core/editor/toggle-list-view`
+				// shortcut callback from being twice.
 				event.preventDefault();
 				closeListView();
 			}
 		},
 		[ closeListView ]
 	);
 
+	const firstRenderCheckRef = useRef( false );
+
+	useEffect( () => {
+		// This extra check avoids duplicate updates of the counter in development
+		// mode (React.StrictMode) or because of potential re-renders triggered
+		// by components higher up the tree.
+		if ( firstRenderCheckRef.current ) {
+			return;
+		}
+		renderCounter++;
+		firstRenderCheckRef.current = true;
```

Comment:
I think it highlights the fragility of the `renderCounter`. React might re-render or mount the component for various reasons. I don't have any better ideas now, but I will try to get back to this later.

P.S. Sorry, I can only do a pass-by comment, at the moment :(

Reply:
Yeah I know. It's a problem that I'm not sure can be addressed with any built-in React feature. For the `ref` check [I took partial inspiration from here](https://github.com/WordPress/gutenberg/blob/4e16551db5990ef74291496e2d87415abc5d1a07/packages/block-library/src/utils/hooks.js#L48-L53).

However, please note that this counter doesn't really need to be an accurate 'count' of the renders. It just needs to understand when it's the first render on first page load. React re-renders wouldn't matter I guess. Not sure about a complete remount but the counter variable is declared and populated outside of the component.

Thanks for having a look, I know you are short on time 🙏🏻